pynamodb/pagination.py

```diff
@@ -190,7 +189,6 @@ def __next__(self) -> _T:
             raise StopIteration
 
         if self._first_iteration:
-            self._first_iteration = False
```

Comment:
My question earlier was around removing this line without a replacement -- you should see the bug through the broken tests

Reply:
Sorry for missing that. I am going to drop this merge request. 
What I was trying to create was the ability for re-tries within a ResultIterator. Does that sound worthy of a feature request to you?

Reply:
Not sure it makes sense for retries to occur within `ResultIterator`, rather than within the lower level connection class. Might be better to open a new issue with the larger context around what you're doing and where existing retries aren't working!